base/mpfr.jl

```diff
@@ -765,12 +765,15 @@ function with_bigfloat_precision(f::Function, precision::Integer)
 end
 
 function string(x::BigFloat)
-    lng::Int32 = 127
+    # In general, the number of decimal places needed to read back the number exactly is
+    # ceil(log(10, 2^precision(x))) or ceil(precision(x) * log10(2))
```

Comment:
I guess I should change the comment which is misleading. You're right in that the number should be 1 higher, but here we only need `ceil(precision(x) * log10(2))`. We call MPFR functions so the number is returned in scientific notation, so I just wanted the most sigfinicand digits except for the first in order to know how decimal places to display after the dot.

I'll change the comment.